flink-end-to-end-tests/flink-plugins-test/src/main/java/org/apache/flink/fs/dummy/DummyFs.java

```
@@ -0,0 +1,166 @@
+
+	private final URI homeDir;
+
+	private final Map<String, byte[]> contents;
+
+	DummyFs(Map<String, String> contents) {
+		this.workingDir = new File(System.getProperty("user.dir")).toURI();
+		this.homeDir = new File(System.getProperty("user.home")).toURI();
+		this.contents = convertToByteArrayMap(contents);
+	}
+
+	// ------------------------------------------------------------------------
+
+	@Override
+	public URI getUri() {
+		return FS_URI;
+	}
+
+	@Override
+	public Path getWorkingDirectory() {
+		return new Path(workingDir);
+	}
+
+	@Override
+	public Path getHomeDirectory() {
+		return new Path(homeDir);
+	}
+
+	@Override
+	public boolean exists(Path f) throws IOException {
+		return getDataByPath(f) != null;
+	}
+
+	@Override
+	public FileStatus[] listStatus(final Path f) throws IOException {
+		byte[] data = getDataByPath(f);
+		if (data == null) {
+			return null;
+		}
+		return new FileStatus[] { new DummyFsFileStatus(f, data.length) };
+	}
+
+	@Override
+	public BlockLocation[] getFileBlockLocations(FileStatus file, long start, long len) throws IOException {
+		return new BlockLocation[] {
+			new LocalBlockLocation(HOSTNAME, file.getLen())
+		};
+	}
+
+	@Override
+	public FileStatus getFileStatus(Path f) throws IOException {
+		byte[] data = getDataByPath(f);
+		if (data == null) {
+			throw new FileNotFoundException("File " + f + " does not exist or the user running "
+				+ "Flink ('" + System.getProperty("user.name") + "') has insufficient permissions to access it.");
+		}
+		return new DummyFsFileStatus(f, data.length);
+	}
+
+	@Override
+	public FSDataInputStream open(final Path f, final int bufferSize) throws IOException {
+		return open(f);
+	}
+
+	@Override
+	public FSDataInputStream open(final Path f) throws IOException {
+		return DummyFsInputStream.create(getDataByPath(f));
+	}
+
+	@Override
+	public boolean delete(final Path path, final boolean recursive) throws IOException {
+		checkNotNull(path, "path is null");
```

nit: I would drop `checkNotNull` for methods that `throw new UnsupportedOperationException();`